contracts/SpokePool.sol

```diff
@@ -1621,6 +1667,118 @@ abstract contract SpokePool is
         }
     }
 
+    // @param relayer: relayer who is actually credited as filling this deposit. Can be differenet from
+    // exclusiveRelayer if passed exclusivityDeadline or if slow fill.
+    function _fillRelayUSS(USSRelayExecutionParams memory relayExecution, address relayer) internal {
+        USSRelayData memory relayData = relayExecution.relay;
+
+        // solhint-disable-next-line not-rely-on-time
+        if (relayData.fillDeadline < block.timestamp) revert ExpiredFillDeadline();
+
+        bytes32 relayHash = relayExecution.relayHash;
+        // If a slow fill for this fill was requested then the relayFills value for this hash will be
+        // FillStatus.RequestedSlowFill. Therefore, if this is the status, then this fast fill
+        // will be replacing the slow fill. If this is a slow fill execution, then the following variable
+        // is trivially false since a slow fill cannot replace itself. We'll emit this param in the FilledRelay
+        // event to assist the Dataworker in knowing when to return funds back to the HubPool that can no longer
+        // be used for a slow fill execution.
+        bool replacedSlowFillExecution = !relayExecution.slowFill &&
+            relayFills[relayExecution.relayHash] == uint256(FillStatus.RequestedSlowFill);
+
+        // @dev This function doesn't support partial fills. Therefore, we associate the relay hash with
+        // an enum tracking its fill status. All filled relays, whether slow or fast fills, are set to the Filled
+        // status. However, we also use this slot to track whether this fill had a slow fill requested. Therefore
+        // we can include a bool in the FilledRelay event making it easy for the dataworker to compute if this
+        // fill was a fast fill that replaced a slow fill and therefore this SpokePool has excess funds that it
+        // needs to send back to the HubPool.
+        if (relayFills[relayHash] == uint256(FillStatus.Filled)) revert RelayFilled();
+        relayFills[relayHash] = uint256(FillStatus.Filled);
+
+        uint256 amountToSend = relayExecution.updatedOutputAmount;
+
+        // This can only happen in a slow fill, where the contract is funding the relay.
```

Comment:
If we want to remove all UBA/dynamic fee logic until later, we should be able to safely remove all this payout adjustment logic, which should free up some bytecode!

Reply:
I think we still need to use payoutAdjustmentPct to rebate the recipient their relayer fee component, right? Because we can't modify the output amount without changing the relayHash

Reply:
Great point. Would it be simpler to just pass it in as an updatedOutputAmount to this function rather than a payout adjustment? Or would that require too many structural changes to the slow relay?

Reply:
I think updatedOutputAmount and payoutAdjustment are the same in terms of calldata and extra data stored in the relay execution struct, right? If so then I'd opt for the former because it's a more intuitive name and you can eliminate the percentage multiplication logic that probably adds gas

Reply:
oh wait nm, updatedOutputAmount is already in the struct! this is def an improvement +1

Reply:
03ba553e4a85d2f4940c2bb7ec216e44dd4ace56